tests/cli/workflow/nf_analysis/test_cli_run.py

```diff
@@ -6,141 +6,160 @@
     assert result.exit_code != EXIT_SUCCESS
     # THEN ERROR log should be printed containing invalid case_id
     assert case_id_does_not_exist in caplog.text
     assert "could not be found" in caplog.text
 
 
+@pytest.mark.parametrize(
+    "context", ["raredisease_context", "rnafusion_context", "taxprofiler_context"]
+)
 def test_without_samples(
     cli_runner: CliRunner,
-    rnafusion_context: CGConfig,
+    context: CGConfig,
     caplog: LogCaptureFixture,
     no_sample_case_id: str,
+    request,
 ):
     """Test command with case_id and no samples."""
     caplog.set_level(logging.ERROR)
+    context = request.getfixturevalue(context)
+
     # GIVEN case-id
     case_id: str = no_sample_case_id
     # WHEN dry running with dry specified
-    result = cli_runner.invoke(run, [case_id, "--dry-run"], obj=rnafusion_context)
+    result = cli_runner.invoke(run, [case_id, "--dry-run"], obj=context)
     # THEN command should NOT execute successfully
     assert result.exit_code != EXIT_SUCCESS
     # THEN warning should be printed that no config file is found
     assert case_id in caplog.text
     assert "no samples" in caplog.text
 
 
+@pytest.mark.parametrize(
+    "context,case_id",
+    [
+        ("raredisease_context", "raredisease_case_id"),
+        ("rnafusion_context", "rnafusion_case_id"),
+        ("taxprofiler_context", "taxprofiler_case_id"),
+    ],
+)
 def test_without_config_dry_run(
     cli_runner: CliRunner,
-    rnafusion_context: CGConfig,
+    context: CGConfig,
     caplog: LogCaptureFixture,
-    rnafusion_case_id: str,
+    case_id: str,
+    request,
 ):
     """Test command dry-run with case_id and no config file."""
     caplog.set_level(logging.ERROR)
+    context = request.getfixturevalue(context)
+
     # GIVEN case-id
-    case_id: str = rnafusion_case_id
+    case_id: str = request.getfixturevalue(case_id)
+
     # WHEN dry running with dry specified
-    result = cli_runner.invoke(run, [case_id, "--from-start", "--dry-run"], obj=rnafusion_context)
+    result = cli_runner.invoke(run, [case_id, "--from-start", "--dry-run"], obj=context)
     # THEN command should execute successfully (dry-run)
     assert result.exit_code == EXIT_SUCCESS
 
 
+@pytest.mark.parametrize(
+    "context,case_id",
+    [
+        ("raredisease_context", "raredisease_case_id"),
+        ("rnafusion_context", "rnafusion_case_id"),
+        ("taxprofiler_context", "taxprofiler_case_id"),
+    ],
+)
 def test_without_config(
     cli_runner: CliRunner,
-    rnafusion_context: CGConfig,
+    context: CGConfig,
     caplog: LogCaptureFixture,
-    rnafusion_case_id: str,
+    case_id: str,
+    request,
 ):
     """Test command with case_id and no config file."""
     caplog.set_level(logging.ERROR)
+    context = request.getfixturevalue(context)
+
     # GIVEN case-id
-    case_id: str = rnafusion_case_id
+    case_id: str = request.getfixturevalue(case_id)
     # WHEN dry running with dry specified
```

Comment:
This is not using dry run

Reply:
Indeed, there is 2 tests: a dry run and a "real" run (that should fail). I updated the docstring